shm/src/Player.cpp

```diff
@@ -0,0 +1,23 @@
+#include "shm/inc/Player.hpp"
+
+Player::Player(std::unique_ptr<Ship> ship, size_t money, size_t availableSpace)
+    : ship_(std::move(ship)), money_(money), availableSpace_(availableSpace) 
+{}
+
+size_t Player::getSpeed() const {
+    return ship_->getSpeed();
+}
+  
+std::shared_ptr<Cargo> Player::getCargo(size_t index) const {
+    return ship_->getCargo(index);
+}
+
+size_t Player::countAvailableSpace() const {
+    return ship_->getCapacity() - 
+            std::accumulate(ship_->getVectorCargo().cbegin(),
+                    ship_->getVectorCargo().cend(),
+                    static_cast<size_t>(0),
+                    [](size_t sum, const auto& cargo) {
+                        return sum += cargo->getAmount();
+                    });
```

Comment:
Maybe we should cache this value? Notice, that this is an expensive function to calculate available space. And calculated value is valid until we add/remove cargo. So it is worth caching it. Think about `std::pair<bool, size_t>` where bool is `true` when value is valid and `false` when invalid. So you can immediately return `size_t` when a flag is `true` and do recalculation only when is `false`. There are also other ways, try to think how you want to avoid unnecessary calculations :)